hoomd/md/pytest/test_angle.py

```diff
@@ -0,0 +1,185 @@
+    harmonic_energies = [0.4112, 0.3427, 0.6854]
+    cosinesq_forces = [-1.29904, 1.7936, 1.58494]
+    cosinesq_energies = [0.375, 0.214466, 0.334936]
+
+    angle_args_forces_and_energies = []
+    for i in range(3):
+        angle_args_forces_and_energies.append(
+            (hoomd.md.angle.Harmonic, harmonic_arg_list[i], harmonic_forces[i],
+             harmonic_energies[i]))
+    for i in range(3):
+        angle_args_forces_and_energies.append(
+            (hoomd.md.angle.Cosinesq, cosinesq_arg_list[i], cosinesq_forces[i],
+             cosinesq_energies[i]))
+    return angle_args_forces_and_energies
+
+
+@pytest.fixture(scope='session')
+def triplet_snapshot_factory(device):
+
+    def make_snapshot(d=1.0,
+                      theta_deg=60,
+                      particle_types=['A'],
+                      dimensions=3,
+                      L=20):
+        theta_rad = theta_deg * (np.pi / 180)
+        s = hoomd.Snapshot(device.communicator)
+        N = 3
+        if s.exists:
+            box = [L, L, L, 0, 0, 0]
+            if dimensions == 2:
+                box[2] = 0
+            s.configuration.box = box
+            s.particles.N = N
+            # shift particle positions slightly in z so MPI tests pass
+            s.particles.position[:] = [
+                [-d * np.sin(theta_rad / 2), d * np.cos(theta_rad / 2), 0.1],
+                [0.0, 0.0, 0.1],
+                [d * np.sin(theta_rad / 2), d * np.cos(theta_rad / 2), 0.1]
+            ]
+            s.particles.types = particle_types
+            if dimensions == 2:
+                box[2] = 0
+                s.particles.position[:] = [[
+                    -d * np.sin(theta_rad / 2), d * np.cos(theta_rad / 2) + 0.1,
+                    0.0
+                ], [0.0, 0.1, 0.0],
+                                           [
+                                               d * np.sin(theta_rad / 2),
+                                               d * np.cos(theta_rad / 2) + 0.1,
+                                               0.0
+                                           ]]
+        return s
+
+    return make_snapshot
+
+
+@pytest.mark.parametrize("angle_and_args", get_angle_and_args())
+def test_before_attaching(angle_and_args):
+    angle, args = angle_and_args
+    angle_potential = angle()
+    angle_potential.params['backbone'] = args
+    for key in args.keys():
+        np.testing.assert_allclose(angle_potential.params['backbone'][key],
+                                   args[key],
+                                   rtol=1e-6)
+
+
+@pytest.mark.parametrize("angle_and_args", get_angle_and_args())
+def test_after_attaching(triplet_snapshot_factory, simulation_factory,
+                         angle_and_args):
+    snap = triplet_snapshot_factory()
+    if snap.exists:
+        snap.angles.N = 1
+        snap.angles.types = ['backbone']
+        snap.angles.typeid[0] = 0
+        snap.angles.group[0] = (0, 1, 2)
```

Comment:
Is `triplet_snapshot_factory` used anywhere else other than `test_angle.py`? If not this this should be in that factory function.